tests/Storefront/Controller/SearchControllerTest.php

```diff
@@ -81,4 +81,76 @@ public function testAvailableFilterReturnsCorrectResponse(string $demoResponse,
 
         $this->assertSame($filterResponse, $expectedFilters);
     }
+
+    public function testFiltersWhichAreNotInTheAvailleFilterResponseAreStillReturned(): void
+    {
+        if (Utils::versionLowerThan('6.3.3.0')) {
+            $this->markTestSkipped('Filter disabling feature was introduced in Shopware 6.3.3.0');
+        }
+
+        $allFiltersResponse = new Xml21Response($this->getMockResponse('XMLResponse/demoResponseWithNoResults.xml'));
+        $responseParser = new Xml21ResponseParser($allFiltersResponse);
+        $availableFilters = $responseParser->getFiltersExtension();
+
+        $response = new Xml21Response($this->getMockResponse('XMLResponse/demoResponseWithAvailableFilters.xml'));
+        $parser = new Xml21ResponseParser($response);
+        $allFilters = $parser->getFiltersExtension();
+
+        $eventMock = $this->getMockBuilder(ProductSearchCriteriaEvent::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $request = new Request();
+        $criteria = new Criteria();
+        $criteria->setExtensions(['flAvailableFilters' => $availableFilters, 'flFilters' => $allFilters]);
+        $eventMock->method('getRequest')->willReturn($request);
+        $eventMock->method('getCriteria')->willReturn($criteria);
+
+        $filterHandler = new FilterHandler();
+        $filterResponse = $filterHandler->handleAvailableFilters($eventMock);
+        $expectedFilters = [
+            'properties' => [
+                'entities' => [
+                    'rating' => [
+                        'max' => 0
```

Comment:
This means only rating is available, others are disabled? It's probably only possible to see that when checking both demo xml responses, right?

Reply:
Also rating is disabled. What I could see from Huzaifas commit messages, is, that we need this in order for Shopware to disable the Rating filter properly.